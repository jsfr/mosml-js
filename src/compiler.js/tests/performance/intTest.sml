prim_val ctimebegin_ : string -> unit = 1 "console.time";
prim_val ctimeend_ : string -> unit = 1 "console.timeEnd";
fun ctimeBegin n = ctimebegin_ n;
fun ctimeEnd n = ctimeend_ n;
fun repeat n f x =
    let val n = ref n
    in  while !n > 0 do (n := !n-1; f x) end;
fun timerep n f s = repeat n (fn x => (ctimeBegin(s);f x; ctimeEnd(s)));

fun recAdd 9500 = ()
  | recAdd x = recAdd (x+1)

fun recSub 0 = ()
  | recSub x = recSub (x-1)

val it = timerep 100 recAdd "recAdd" 0
val it = timerep 100 recSub "recSub" 9500