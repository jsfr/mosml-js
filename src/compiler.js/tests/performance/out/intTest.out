beautified intTest.js
==============================JS BEGIN==============================
recAdd: 4ms
recAdd: 2ms
recAdd: 2ms
recAdd: 1ms
recAdd: 1ms
recAdd: 2ms
recAdd: 2ms
recAdd: 1ms
recAdd: 1ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 1ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 0ms
recAdd: 2ms
recAdd: 0ms
recAdd: 1ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 2ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 2ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 0ms
recAdd: 1ms
recAdd: 2ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 2ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 2ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 2ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 2ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 2ms
recAdd: 0ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 1ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 2ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 4ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recAdd: 1ms
recAdd: 0ms
recSub: 1ms
recSub: 1ms
recSub: 0ms
recSub: 2ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 0ms
recSub: 0ms
recSub: 0ms
recSub: 2ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 1ms
recSub: 0ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 0ms
recSub: 2ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 1ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 2ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 0ms
recSub: 2ms
recSub: 1ms
recSub: 0ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 1ms
recSub: 1ms
recSub: 0ms
recSub: 0ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
recSub: 0ms
recSub: 2ms
recSub: 0ms
recSub: 1ms
recSub: 1ms
recSub: 0ms
recSub: 1ms
==============================JS END================================
